### PR TITLE
Pytorch Feature Tutorials Refactor

### PR DESCRIPTION
**Pull request by lior-dikstein**
## Pull Request Description:
Refactored Pytorch feature notebooks for consistency. Standardized function usage across notebooks. Updated model examples to use pretrained models from torchvision with the ImageNet validation set as the representative dataset (except for the pruning notebook).

## Checklist before requesting a review:
- [x] I set the appropriate labels on the pull request.
- [x] I have added/updated the release note draft (if necessary).
- [x] I have updated the documentation to reflect my changes (if necessary).
- [x] All function and files are well documented. 
- [x] All function and classes have type hints.
- [x] There is a licenses in all file.
- [x] The function and variable names are informative. 
- [x] I have checked for code duplications.
- [x] I have added new unittest (if necessary).

### CODE REVIEW

File: tutorials/notebooks/mct_features_notebooks/pytorch/example_pytorch_post_training_quantization.ipynb [hunk 1]
```diff
@@ -0,0 +1,470 @@
...
+    "\n",
+    "## Overview\n",
+    "This quick-start guide explains how to use the **Model Compression Toolkit (MCT)** to quantize a PyTorch model. We will load a pre-trained model and  quantize it using the MCT with **Post-Training Quatntization (PTQ)**. Finally, we will evaluate the quantized model and export it to an ONNX file.\n",
+    "\n",
+    "## Summary\n",
+    "In this tutorial, we will cover:\n",
+    "\n",
+    "1. Loading and preprocessing ImageNet’s validation dataset.\n",
+    "2. Constructing an unlabeled representative dataset.\n",
+    "3. Hardware-Friendly Post-Training Quantization using MCT.\n",
```
Comment (ofirgo) [inline tutorials/notebooks/mct_features_notebooks/pytorch/example_pytorch_post_training_quantization.ipynb:21]: I think we can remove the "Hardware-friendly" term from this tutorial, the weights thresholds are no longer POT (in the default TPC) and this is not important in the scope of the tutorial and its purpose.

File: tutorials/notebooks/mct_features_notebooks/pytorch/example_pytorch_post_training_quantization.ipynb [hunk 2]
```diff
@@ -0,0 +1,470 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4f5fa4a2",
+   "metadata": {
+    "id": "4f5fa4a2"
+   },
+   "outputs": [],
+   "source": [
+    "print(quantized_model)"
```
Comment (ofirgo) [inline tutorials/notebooks/mct_features_notebooks/pytorch/example_pytorch_post_training_quantization.ipynb:282]: remove

**Review by ofirgo — CHANGES_REQUESTED** [review]

File: tutorials/notebooks/mct_features_notebooks/pytorch/example_pytorch_data_generation.ipynb [hunk 3]
```diff
@@ -245,206 +201,187 @@
...
-    "            batch_avg_top_1, correct_inds = accuracy(outputs=predicted_batch, labels=target_batch)\n",
-    "            acc_top1 += batch_avg_top_1[0].item()\n",
-    "    \n",
-    "    \n",
-    "            batch_cntr += 1\n",
-    "            if batch_cntr > iterations:\n",
-    "                break\n",
-    "    acc_top1 /= iterations\n",
-    "    return acc_top1    "
+    "batches_inds = np.random.choice(len(generated_images),\n",
```
[the remark this answers is not in the document]
Comment (lior-dikstein) [reply tutorials/notebooks/mct_features_notebooks/pytorch/example_pytorch_data_generation.ipynb:256]: It can, but its best practice to do it this way to get the best results

File: tutorials/notebooks/mct_features_notebooks/pytorch/example_pytorch_mixed_precision_ptq.ipynb [hunk 4]
```diff
@@ -0,0 +1,515 @@
...
+    "from torchvision.models import mobilenet_v2, MobileNet_V2_Weights\n",
+    "from torchvision.datasets import ImageNet\n",
+    "import numpy as np\n",
+    "import random"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "We will set a random seed to ensure reproducibility of results."
```
[the remark this answers is not in the document]
Comment (lior-dikstein) [reply tutorials/notebooks/mct_features_notebooks/pytorch/example_pytorch_mixed_precision_ptq.ipynb:72]: I erased it.

**Review by ofirgo — APPROVED** [review]
